app/uk/gov/hmrc/cgtpropertydisposals/service/dms/DmsSubmissionPoller.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.service.dms
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorSystem
+import cats.data.EitherT
+import cats.instances.future._
+import cats.instances.int._
+import cats.syntax.eq._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import uk.gov.hmrc.cgtpropertydisposals.models.Error
+import uk.gov.hmrc.cgtpropertydisposals.service.DmsSubmissionService
+import uk.gov.hmrc.cgtpropertydisposals.service.dms.DmsSubmissionPoller.OnCompleteHandler
+import uk.gov.hmrc.cgtpropertydisposals.util.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.workitem.{Failed, PermanentlyFailed, Succeeded, WorkItem}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+@Singleton
+class DmsSubmissionPoller @Inject() (
+  actorSystem: ActorSystem,
+  dmsSubmissionService: DmsSubmissionService,
+  dmsSubmissionPollerContext: DmsSubmissionPollerExecutionContext,
+  servicesConfig: ServicesConfig,
+  onCompleteHandler: OnCompleteHandler
+)(implicit
+  executionContext: DmsSubmissionPollerExecutionContext
+) extends Logging {
+
+  private val jitteredInitialDelay: FiniteDuration = FiniteDuration(
+    servicesConfig.getDuration("dms.submission-poller.initial-delay").toNanos,
+    TimeUnit.NANOSECONDS
+  ) + FiniteDuration(
+    Random.nextInt((servicesConfig.getDuration("dms.submission-poller.jitter-period").toSeconds.toInt + 1)).toLong,
+    TimeUnit.NANOSECONDS
```

Comment:
shouldn't this be `TimeUnit.SECONDS`?

Reply:
or should the `toSeconds` be `toNanos` above?

Reply:
fixed - time units should be SECONDS.